lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1912,6 +1908,23 @@ func (d *zfs) SetVolumeQuota(vol Volume, size string, allowUnsafeResize bool, op
 	return nil
 }
 
+// GetVolumeDiskPath returns the location of a root disk block device.
+func (d *zfs) tryGetVolumeDiskPathFromDataset(ctx context.Context, dataset string) (string, error) {
+	for {
+		select {
+		case <-ctx.Done():
+			return "", fmt.Errorf("Failed to locate zvol for %q: %v", dataset, ctx.Err())
```

Comment:
Is zvol the right term here? A zvol is used for a block volume (with or without filesystem on it) and a dataset is used for a zfs filesystem. Seems the terminology maybe confused in certain scenarios here.

Reply:
This should be correct, since the function will try to locate a device path of the block device. It calls `getVolumeDiskPathFromDataset` which check for `/dev/zvol/<dataset>` path and if not found, tries to located the corresponding device on path `/dev/zd*`.